contracts/LiquidityBridgeContract.sol

```diff
@@ -381,23 +401,41 @@ contract LiquidityBridgeContract {
     }
 
     function registerPegOut(
-        Quote memory quote,
+        PegOutQuote memory quote,
         bytes memory signature
     ) public noReentrancy payable {
-        bytes32 quoteHash = validateAndHashQuote(quote);
+        bytes32 quoteHash = validateAndHashPegOutQuote(quote);
 
         require(SignatureValidator.verify(quote.liquidityProviderRskAddress, quoteHash, signature), "LBC: Invalid signature");
-        require(quote.timeForDeposit < block.timestamp, "LBC: Block height overflown");
-        require(processedQuotes[quoteHash] != 2, "LBC: Quote already pegged out");
-        processedQuotes[quoteHash] = PROCESSED_QUOTE_CODE;
+        require(quote.depositDateLimit < block.timestamp, "LBC: Block height overflown");
+        require(processedPegOutQuotes[quoteHash] != 2, "LBC: Quote already pegged out");
+        processedPegOutQuotes[quoteHash] = PROCESSED_QUOTE_CODE;
 
-        uint256 valueToTransfer = quote.value + quote.callFee;
+        uint256 valueToTransfer = quote.valueToTransfer + quote.fee;
         require(msg.value == valueToTransfer, "LBC: msg value doesnt match quote");
         require(address(this).balance >= valueToTransfer, "LBC: Not enough funds");
 
         increasePegOutBalance(quote.rskRefundAddress, valueToTransfer);
 
-        emit PegOut(msg.sender, quote.value, quoteHash, processedQuotes[quoteHash]);
+        emit PegOut(msg.sender, quote.valueToTransfer, quoteHash, processedPegOutQuotes[quoteHash]);
+    }
+
+    function refundPegOut(
+        PegOutQuote memory quote,
+        bytes32 btcTxHash,
+        bytes32 btcBlockHeaderHash,
+        uint256 partialMerkleTree,
+        bytes32[] memory merkleBranchHashes
+    ) public noReentrancy {
+        bytes32 quoteHash = validateAndHashPegOutQuote(quote);
+        require(processedPegOutQuotes[quoteHash] == 2, "LBC: Quote not processed");
+        require(block.timestamp <= quote.expireDate, "LBC: Quote expired by date");
+        require(block.number <= quote.expireBlocks, "LBC: Quote expired by blocks");
+        require(msg.sender == quote.lpAddress, "LBC: Wrong sender");
+        require(bridge.getBtcTransactionConfirmations(btcTxHash, btcBlockHeaderHash, partialMerkleTree, merkleBranchHashes) >= quote.transferConfirmations, "LBC: Don't have required confirmations");
+        payable(address(uint160(uint256(quote.derivationAddress)))).transfer(quote.valueToTransfer);
+        decreasePegOutBalance(quote.rskRefundAddress, quote.valueToTransfer);
+        delete processedPegOutQuotes[quoteHash];
```

Comment:
shouldn't we keep track of this somehow?

Reply:
It's tracked on the blockchain, you always can look at historical data and saw that value. Create another struct to track this information will cause an increase in the gas of the transaction.

Reply:
you're right. good point.